### PR TITLE
Feature User UI UX. Auth logic.

### PR DESCRIPTION
**Pull request by vkhoroshchak**

### CODE REVIEW

File: app/auth.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    form = RegisterForm(request)
+    await form.load_data()
+    if await form.is_valid():
+        try:
+            user = models.BaseUserCreate(email=form.username, password=form.password)
+            await user_manager.create(user, safe=True, request=request)
+        except UserAlreadyExists:
+            return templates.TemplateResponse("signup.html", form.__dict__)
+        except InvalidPasswordException:
+            return templates.TemplateResponse("signup.html", form.__dict__)
```
Comment (mranch) [inline app/auth.py]:
```suggestion
        except (UserAlreadyExists, InvalidPasswordException):
            return templates.TemplateResponse("signup.html", form.__dict__)
```

File: app/db.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+load_dotenv()
+
+DATABASE_URL = os.getenv("DATABASE_URL")
+
+client = motor.motor_asyncio.AsyncIOMotorClient(
+    DATABASE_URL, uuidRepresentation="standard"
+)
+db = client[os.getenv("MONGO_DB")]
+collection = db["users"]
```
Comment (mranch) [inline app/db.py:21]: Can this raise KeyError in any case? If so, please use `get` or wrap it in try-except block

File: app/db.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+
+def get_user_db():
+    yield MongoDBUserDatabase(UserDB, collection)
+
+
+class ReportHistoryManager:
+    report_history_collection = db["report_history"]
```
Comment (mranch) [inline app/db.py:29]: Can this raise KeyError in any case? If so, please use get or wrap it in try-except block

File: app/report_history.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            {
+                "request": request,
+                "report_history": await ReportHistoryManager(user).retrieve_records(),
+                "user_name": user.email,
+                "user_id": user.id,
+             }
+        )
+    else:
+        logger.info("Unauthorized user, redirecting to sign in page")
+        return RedirectResponse(url="/auth/signin", status_code=302)
```
Comment (mranch) [inline app/report_history.py]: Should status code be 401 (Unauthorized) instead of 302(Found) ?

File: app/users.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        self.errors: List = []
+        self.username: Optional[str] = None
+        self.password: Optional[str] = None
+        self.confirm_password: Optional[str] = None
+
+    async def load_data(self):
+        form = await self.request.form()
+        self.username = form.get(
+            "username"
+        )
```
Comment (mranch) [inline app/users.py]: Can this be in one line or this can be modified in the future with more arguments?

**Review by mranch — COMMENTED** [review]
Left several comments, still need to review JS files and those after them

File: app/db.py (hunk 2 above)
Comment (vkhoroshchak) [reply app/db.py:21]: No, It will create the collection ("users") if it does not exist

File: app/db.py (hunk 3 above)
Comment (vkhoroshchak) [reply app/db.py:29]: Same

File: app/report_history.py (hunk 4 above)
Comment (vkhoroshchak) [reply app/report_history.py]: Yes, it's better describes the redirection reason

**Review by mranch — APPROVED** [review]

**Pull request merged**